cmd/bridge/server/server.go

```diff
@@ -0,0 +1,91 @@
+package server
+
+import (
+	"context"
+	"crypto/tls"
+	"flag"
+	"net"
+	"net/http"
+
+	"github.com/openshift/console/pkg/flags"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/certwatcher"
+)
+
+type Server struct {
+	listenURL string
+	certFile  string
+	keyFile   string
+}
+
+func NewServer(fs *flag.FlagSet) *Server {
+	s := &Server{}
+	fs.StringVar(&s.listenURL, "listen", "http://0.0.0.0:9000", "")
+	fs.StringVar(&s.certFile, "tls-cert-file", "", "TLS certificate. If the certificate is signed by a certificate authority, the certFile should be the concatenation of the server's certificate followed by the CA's certificate.")
+	fs.StringVar(&s.keyFile, "tls-key-file", "", "The TLS certificate key.")
+	return s
+}
+
+func (s *Server) Serve(handler http.Handler) error {
+	// Context to clean up on exit
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	url, err := flags.ValidateFlagIsURL("listen", s.listenURL, false)
+	flags.FatalIfFailed(err)
+
+	switch url.Scheme {
+	case "http":
+	case "https":
+		flags.FatalIfFailed(flags.ValidateFlagNotEmpty("tls-cert-file", s.certFile))
+		flags.FatalIfFailed(flags.ValidateFlagNotEmpty("tls-key-file", s.keyFile))
+	default:
+		flags.FatalIfFailed(flags.NewInvalidFlagError("listen", "scheme must be one of: http, https"))
+	}
+
+	klog.Infof("Binding to %s...", url.Host)
+	httpsrv := &http.Server{Handler: handler}
+	listener, err := s.listen(ctx, url.Scheme, url.Host)
+	if err != nil {
+		return err
+	}
+	defer listener.Close()
+
+	go func() {
+		<-ctx.Done()
+		klog.Infof("Shutting down server...")
+		if err = httpsrv.Shutdown(ctx); err != nil {
+			klog.Errorf("Error shutting down server: %v", err)
+		}
+	}()
+
+	// Cancel context and close listener on Serve return
+	return httpsrv.Serve(listener)
+}
+
+func (s *Server) listen(ctx context.Context, scheme, host string) (net.Listener, error) {
```

Comment:
I would suggest to name the method after what actually returns
```suggestion
func (s *Server) getListener(ctx context.Context, scheme, host string) (net.Listener, error) {
```

Reply:
I was following the naming of `tls.Listen` and `net.Listen` since this func just wraps those. I can change if needed though.